soccer/Robot.hpp

```diff
@@ -430,19 +470,23 @@ class OurRobot: public Robot {
 	RobotMask _self_avoid_mask, _opp_avoid_mask;  /// masks for obstacle avoidance
 	float _avoidBallRadius; /// radius of ball obstacle
 
+	Planning::MotionCommand _motionCommand;
+	Planning::MotionCommand::CommandType _lastCommandType;
 	MotionConstraints _motionConstraints;
 
-	Planning::RRTPlanner *_planner;	/// single-robot RRT planner
+	std::shared_ptr<Planning::PathPlanner> _planner;	/// single-robot RRT planner
```

Comment:
Not sure yet if we'll use the same one for multiple robots. Shared pointers
are a safe bet anyways without any real performance  disadvantages on our
systems.
On Jul 21, 2015 4:57 PM, "Justin Buchanan" <notifications@github.com> wrote:

> In soccer/Robot.hpp
> <https://github.com/RoboJackets/robocup-software/pull/373#discussion_r35082039>
> :
>
> >  	MotionConstraints _motionConstraints;
> >
> > -	Planning::RRTPlanner *_planner;	/// single-robot RRT planner
> > +	std::shared_ptr<Planning::PathPlanner> _planner;	/// single-robot RRT planner
>
> Should this be a unique_ptr instead?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/RoboJackets/robocup-software/pull/373/files#r35082039>
> .
>